### PR TITLE
Make form save messages distinguish between incomplete/complete saves

### PR DESCRIPTION
**Pull request by phillipm**
Meryn and the PPIUD team brought up an interesting point that users were getting confused by the fact that when you save a form as incomplete via the drop down menu, you get the same message as when you save a form as complete.

This PR makes it so when you save a form using the drop down menu, you get:
> Form saved as incomplete

And when you complete a form at the end-of-form nav, it shows:
> Form saved and ready for submission

Perhaps the "ready for submission" part is confusing because it makes it seem like the user has something they need to do?

screen1

screen

### CODE REVIEW

**Comment by phillipm** [comment]
@amsagoff wondering, do you have suggestions on the string text I've chosen?

**Comment by amsagoff** [comment]
@phillipm agree "ready for submission" implies some action from the user. What if we kept that as "form successfully saved" and instead changed the submission text to "form successfully completed"?

Though ultimately I think meryn's issue was with the end of form UI being confusing.

**Comment by amsagoff** [comment]
@phillipm to make sure i understand - when you're on the last question and you save the form, does it get saved as incomplete or saved and submitted or...what

**Comment by phillipm** [comment]
@amsagoff if the bar is green but you save the form using the drop-down menu, it still gets saved as incomplete. This feels wrong to me, but that is how it works currently.

I like your text suggestions, will update

**Comment by amsagoff** [comment]
That actually feels right to me - in that case, I have a new suggestion: why doesn't the text say "form saved as incomplete" if that's what's really happening?

File: app/src/org/odk/collect/android/tasks/SaveToDiskTask.java [hunk 1]
```diff
@@ -373,13 +379,17 @@ private int validateAnswers(boolean markCompleted, boolean fireTriggerables) {
                     saveStatus =
                             FormEntryActivity.mFormController.checkCurrentQuestionConstraint();
                 }
-                if (markCompleted && saveStatus != FormEntryController.ANSWER_OK) {
-                    return saveStatus;
+                if (markCompleted &&
+                        saveStatus != FormEntryController.ANSWER_OK && // not technically needed
```
Comment (amstone326) [inline app/src/org/odk/collect/android/tasks/SaveToDiskTask.java]: looks like you already noted this, but why leave this line in?

Comment (phillipm) [inline app/src/org/odk/collect/android/tasks/SaveToDiskTask.java]: Not sure, just took it out.

**Pull request merged**